.projenrc.js

```diff
@@ -1,6 +1,41 @@
-const { readFileSync, writeFileSync } = require("fs");
-const { join } = require("path");
-const { typescript } = require("projen");
+const { readFileSync, writeFileSync, mkdirSync, chmodSync } = require("fs");
+const { join, dirname } = require("path");
+const { typescript, Component } = require("projen");
+
+/**
+ * Adds githooks into the /git/hooks folder during projen synth.
+ *
+ * @see https://git-scm.com/docs/githooks
+ *
+ * githooks: {
+ *    preCommit: ["lint-staged"] // array of bash commands
+ * }
+ */
+class GitHooksComponent extends Component {
```

Comment:
Can't you use the TextFile component instead of this?

Reply:
Could, yeah, didn't know it existed. Wanted to play with supporting githooks as component instead of just making the file.

TextFile doesn't have chmod option though.